src/BaseRepository.php

```diff
@@ -130,18 +130,32 @@ protected function _select(array $condition, array $params = []): BaseCollection
 	}
 
 	/**
-	 * @param array $condition
-	 * @param array $params
-	 * @return BaseEntity
+	 * Selects the fields of the first row as array
+	 *
 	 * @throws NotFoundException
+	 *
+	 * @return array The resulted fields as array
 	 */
-	protected function _selectOne(array $condition, array $params = []): BaseEntity
+	protected function _selectFirstRowAsArray(array $condition, array $params = []): array
 	{
 		$fields = $this->db->selectFirst(static::$table_name, [], $condition, $params);
+
 		if (!$this->db->isResult($fields)) {
 			throw new NotFoundException();
 		}
 
+		return $fields;
+	}
+
+	/**
+	 * @deprecated 2025.05 Use `\Friendica\BaseRepository::_selectFirstRowAsArray()` instead
```

Comment:
Same question, why moving back from structured objects back to anonymous arrays?

Reply:
Same answer as with `BaseCollection::map()`: `BaseRepository::_selectOne()` return `BaseEntity` instance, but calling classes requires the entities as a specific childclass. So instead check the result for the correct class type, I select the first row as array and call the specific factory myself where I know that the result is the specific childclass.

```diff
// in src/Navigation/Notifications/Repository/Notification.php
	private function selectOne(array $condition, array $params = []): NotificationEntity
	{
-		return parent::_selectOne($condition, $params);
+		$fields = $this->_selectFirstRowAsArray( $condition, $params);
+
+		return $this->factory->createFromTableRow($fields);
	}
```

Reply:
Same answer as well, the `static` return type isn't available, but in practice `_selectOne` always returns a child class object thanks to the `$this->factory` property which will never create a naked `BaseEntity` object.